### ISSUE TITLE
Avoid Recompiling Crates

### ISSUE THREAD

**Issue by nixpulvis** [neon-bindings/neon#270]
When I run `neon build` it will always compiles every crate.

**Comment by nixpulvis** [comment]
This may be a more general version of #242.

**Comment by dherman** [comment]
@nixpulvis Awesome, I've been looking to reproduce! Can you provide me as much info as possible about your build environment?

- OS?
- Node version?
- Rust version?
- Electron vs normal Node?
- Contents of your `native/artifacts.json` file?
- Ideally the full contents of the projects, if you're willing/able to share the source

Thanks!

**Comment by nixpulvis** [comment]
@dherman well this is embarrassing... I've updated a few system dependencies since I opened this issue, and now I can't reproduce it either!

FWIW I'm now running:

- `Linux x86_64 4.13.11-1-ARCH`
- `node v8.8.1`
- rustc 1.22.0-nightly (4c053db23 2017-10-22)
- I haven't played with Electron yet.

Sorry for not including this info when I opened the issue, because we might have stood a chance then. I really wish I had an OS that let me rewind :P
